posts/2020-05-15-five-years-of-rust.md

```diff
@@ -0,0 +1,508 @@
+libraries built with different editions to seamlessly work together.
+
+- **Non-Lexical Lifetimes** A huge improvement to Rust's borrow checker,
+  allowing it to accept more verifiable safe code.
+- **Module System Improvements** Large UX improvements to how we define and
+  use modules.
+- **Const Functions** Const functions allow you to run and evaluate Rust code
+  at compile time.
+- **Rustfmt 1.0** A new code formatting tool built specifically for Rust.
+- **Clippy 1.0** Rust's linter for catching common mistakes. Clippy makes it a lot
+  easier to make sure that your code is not only safe but correct.
+- **Rustfix** With all the syntax changes, we knew we wanted to provide the
+  tooling to make the transition as easy as possible. Now when changes are
+  required to Rust's syntax they're just a `cargo fix` away from being resolved.
+
+#### 2019
+
+**[1.34] — Alternative Crate Registries** As Rust is used more and more in
+production, there is a greater need to be able to host and use your projects
+in non-public spaces, while cargo has always allowed remote git dependencies,
+with Alternative Registries your organisation can easily build and share your
+own registry of crates that can be used in your projects like they were
+on crates.io.
+
+**[1.39] — Async/Await** The stabilisation of the async/await keywords for
+handling Futures was one of the major milestones to making async programming
+in Rust a first class citizen. Even just six months after its release the
+async programming has blossomed into a diverse and performant ecosystem.
+
+[1.2]: https://blog.rust-lang.org/2015/08/06/Rust-1.2.html
+[1.3]: https://blog.rust-lang.org/2015/09/17/Rust-1.3.html
+[1.4]: https://blog.rust-lang.org/2015/10/29/Rust-1.4.html
+[1.5]: https://blog.rust-lang.org/2015/12/10/Rust-1.5.html
+[1.6]: https://blog.rust-lang.org/2016/01/21/Rust-1.6.html
+[1.10]: https://blog.rust-lang.org/2016/07/07/Rust-1.10.html
+[1.12]: https://blog.rust-lang.org/2016/09/29/Rust-1.12.html
+[1.13]: https://blog.rust-lang.org/2016/11/10/Rust-1.13.html
+[1.14]: https://blog.rust-lang.org/2016/12/22/Rust-1.14.html
+[1.15]: https://blog.rust-lang.org/2017/02/02/Rust-1.15.html
+[1.17]: https://blog.rust-lang.org/2017/04/27/Rust-1.17.html
+[1.20]: https://blog.rust-lang.org/2017/08/31/Rust-1.20.html
+[1.24]: https://blog.rust-lang.org/2018/02/15/Rust-1.24.html
+[1.26]: https://blog.rust-lang.org/2018/05/10/Rust-1.26.html
+[1.28]: https://blog.rust-lang.org/2018/08/02/Rust-1.28.html
+[1.31]: https://blog.rust-lang.org/2018/12/06/Rust-1.31-and-rust-2018.html
+[1.34]: https://blog.rust-lang.org/2019/04/11/Rust-1.34.0.html
+[1.39]: https://blog.rust-lang.org/2019/11/07/Rust-1.39.0.html
+
+## Error Diagnostics
+
+One thing that we haven't mentioned much is how much Rust's error messages and
+diagnostics have improved since 1.0. But they have improved so much over these
+past five years, looking at older error messages now feels like looking a
+different language.
```

Comment:
The "but" part here made me stumble while reading. It's not really a contrast to the previous point, but an affirmation! I'd probably just drop that part, so have something like

> One thing that we haven't mentioned much is how much Rust's error messages and
diagnostics have improved since 1.0. Looking at older error messages now feels like looking a
different language.